From: Andrea Azzarone <andrea.azzarone@canonical.com>
Date: Mon, 8 Oct 2018 13:09:10 +0100
Subject: region: Autodisconnect IBusBus "connected" handler

Use g_signal_connect_object to ensure handler is not run after object is
destroyed.

Fixes: GNOME/gnome-control-center#223
Origin: upstream, commit:995a945099388f07c4188b77046095195f8074c0
---
 panels/region/cc-region-panel.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/panels/region/cc-region-panel.c b/panels/region/cc-region-panel.c
index b20106d..1061466 100644
--- a/panels/region/cc-region-panel.c
+++ b/panels/region/cc-region-panel.c
@@ -1319,8 +1319,9 @@ setup_input_section (CcRegionPanel *self)
                 if (ibus_bus_is_connected (self->ibus))
                         fetch_ibus_engines (self);
                 else
-                        g_signal_connect_swapped (self->ibus, "connected",
-                                                  G_CALLBACK (fetch_ibus_engines), self);
+                        g_signal_connect_object (self->ibus, "connected",
+                                                 G_CALLBACK (fetch_ibus_engines), self,
+                                                 G_CONNECT_SWAPPED);
         }
         maybe_start_ibus ();
 #endif
